Ardupilog.m

```diff
@@ -157,6 +157,8 @@ function delete(obj)
             % Construct the LineNo for the whole log
             LineNo_ndx_vec = sort(vertcat(obj.valid_msgheader_cell{:,2}));
             LineNo_vec = [1:length(LineNo_ndx_vec)]';
+            % Save the last LineNo as the number of messages in the log
+            obj.numMsgs = LineNo_vec(end);
```

Comment:
Yeah, these steps sound ideal:
1. Add totalNumMsgs
2. Keep numMsgs
3. Add the used message filter

Will you push an update commit? (remember to pull the branch locally, as I did a rebase and re-wrote history for this branch)